source/helpers/CaseDataConverter.js

```diff
@@ -97,8 +97,13 @@ export const convertAnswersToArray = (data, formQuestions) => {
               const dynamicTagRegex = new RegExp('.+:x');
               newTags = tags.map((tag) => {
                 if (dynamicTagRegex.test(tag)) {
-                  const strArray = tag.split(':');
-                  return `${strArray?.[0] || tag}:${childFieldId}`;
+                  const tagStringList = tag.split(':');
+                  const tagStringIndex = tagStringList.findIndex('x');
+                  if (tagStringIndex !== -1) {
```

Comment:
Prefer writing the if-statment without `!`, because using `!` makes you think more than necessary.

Perhaps this is easier to read:
```js
if (tagStringIndex >= 0) { ... }
```